src/main/java/fr/insalyon/creatis/gasw/execution/GaswSubmit.java

```diff
@@ -85,46 +93,65 @@ public GaswSubmit(GaswInput gaswInput, GaswMinorStatusServiceGenerator minorStat
      *
      * @return
      * @throws GaswException
+     * @throws IOException 
      */
-    protected String generateScript() throws GaswException {
-
-        String script = ScriptGenerator.getInstance().generateScript(
-                gaswInput, minorStatusServiceGenerator);
-
-        return publishScript(gaswInput.getExecutableName(), script);
-    }
-
-    /**
-     *
-     * @param symbolicName Symbolic name of the execution.
-     * @param script Generated script to be saved in a file.
-     * @return Name of the script file.
-     */
-    private String publishScript(String symbolicName, String script) {
-
-        try {
-            File scriptsDir = new File(GaswConstants.SCRIPT_ROOT);
-            if (!scriptsDir.exists()) {
-                scriptsDir.mkdir();
+    protected String generateScript() throws GaswException, IOException {
```

Comment:
This should not throw IOEXception
catch them and retrow a GaswException if necessary